app/models/content/post.rb

```diff
@@ -46,7 +46,7 @@ def set_default_date
     end
 
     def reset_highlighted_flag
-      return true unless highlighted
+      return true unless highlighted && highlighted_changed?
```

Comment:
or maybe we should just conditionally invoke that callback
```
before_save :reset_highlighted_flag, if: -> { highlighted && highlighted_changed? }
```
what do you think @agnessa ?